cap_anndata/backed_df.py

```diff
@@ -31,10 +35,64 @@ def __setitem__(self, key, value) -> None:
         return super().__setitem__(key, value)
 
     @classmethod
-    def from_df(cls, df: pd.DataFrame, column_order: List[str] = None):
+    def from_df(cls, df: pd.DataFrame, column_order: List[str] = None) -> Self:
         if column_order is None:
             column_order = df.columns.to_numpy()
 
         new_inst = cls(df)
         new_inst.column_order = column_order
         return new_inst
+
+    def join(self, other: Any, on=None, how="left", lsuffix="", rsuffix="", sort=False):
+        result = super().join(
+            other, on=on, how=how, lsuffix=lsuffix, rsuffix=rsuffix, sort=sort
+        )
+        if isinstance(other, CapAnnDataDF):
+            new_columns = [
+                col for col in other.column_order if col not in self.column_order
+            ]
+        else:
+            new_columns = [col for col in other.columns if col not in self.column_order]
+        column_order = np.append(self.column_order, new_columns)
+        return self.from_df(result, column_order=column_order)
+
+    def merge(
+        self,
+        right,
+        how="inner",
+        on=None,
+        left_on=None,
+        right_on=None,
+        left_index=False,
+        right_index=False,
+        sort=False,
+        suffixes=("_x", "_y"),
```

Comment:
could we remove it from signature? (to prevent case where original pandas value changed in future)